cmd/git-sync/main.go

```diff
@@ -700,6 +700,13 @@ func addWorktreeAndSwap(ctx context.Context, gitRoot, dest, branch, rev string,
 		return err
 	}
 
+	// Check for rev after fetch
+	_, err := revIsHash(ctx, hash, gitRoot)
```

Comment:
Checking if git can identify the remote hash makes so much sense here. Thanks for the fix.